algonaut_transaction/src/api_model.rs

```diff
@@ -567,6 +593,99 @@ fn int_to_application_call_on_complete(
     }
 }
 
+fn api_box_references_from_box_references(
```

Comment:
We could encode these functions as type conversions, here a gist with the main idea (could be further improved) https://gist.github.com/manuelmauro/0da03cde317596e064e2017c891edf81

Reply:
:) I was not super happy with `BoxesInfo` and `ApiBoxesInfo` and I wonder if just moving their code into `impl TryFrom<Transaction> for ApiTransaction`and `impl TryFrom<ApiTransaction> for Transaction` would be better (probably no, since we use the conversion also in the testing and we would need to duplicate that).
Also, the trait `AppCallMetadata` sounds like a good idea and it probably deserves a place somewhere else, but I'd leave it there for now and see if it pops up again in the future.

In any case, I am happy to merge the PR whenever you want. I'd like to fix the SDK testing CI in the near future to have a better coverage, but for now the test you wrote look good to me!

Reply:
Would it make sense to have `fn from_box_reference(box_, current_app_id, foreign_apps)` in `impl ApiBoxReference` and a counterpart in `impl BoxReference` to convert from an `ApiBoxReference`? Probably not, since they're located in different modules and I'm not sure how rusty that would be.

I think it'd be okay to merge & use what we have now. What I like is that `BoxesInfo` and `ApiBoxesInfo` is used only internally and doesn't leak to the caller. Therefore, it can be refactored in the future if a better idea pops up.

Reply:
Sorry, I missed this for some reason! Gonna merge it then!